src/commandline_argv.cpp

```diff
@@ -65,8 +65,8 @@ std::vector<std::string> win32_read_argv(const std::string& input)
 	std::vector<std::string> res;
 
 	while(win32_parse_single_arg(start, end, buffer)) {
-		res.emplace_back();
-		res.back().swap(buffer);
+		auto& back = res.emplace_back();
+		back.swap(buffer);
```

Comment:
I feel this is one to leave to the compiler's optimiser.

Reply:
to me it feels more natural to use that returned value, instead of obtaining it again via another function call (inlined for sure, but why drop the return?)

Reply:
reverted this one

Reply:
As of C++17 (min we support) you can do `res.emplace_back().swap(buffer)`

Reply:
> As of C++17 (min we support) you can do `res.emplace_back().swap(buffer)`

I wanted to do that, should I?

Reply:
Yes please. Apologies for saying no originally, seeing the reviews from Soliton and Vultraz has made me realise that I was wrong.

Reply:
it's alright, on it